vscode/microsoft-kiota/src/kiotaInterop/kiotaInstall.ts

```diff
@@ -34,15 +34,18 @@ export async function ensureKiotaIsPresent(context: vscode.ExtensionContext) {
                 title: vscode.l10n.t("Downloading kiota...")
 
             }, async (progress, _) => {
-                const online = await isOnline();
-                if (!online) {
-                    await vscode.window.showErrorMessage(
-                        vscode.l10n.t("Downloading kiota requires an internet connection. Please check your connection and try again.")
-                    );
-                    return;
-                }
-                await runIfNotLocked(context, async () => {
-                    try {
+                await (async () => {
```

Comment:
why do we need this anonymous function wrapper? we were already in an async function here.

Reply:
~Because commonjs imports does not support the earlier import of `import isOnline from 'is-online'`.~ Seems to have self-resolved with the latest NPM updates. Reverted back to original.

Reply:
My bad, actually it's not resolved. The reason for the change to a dynamic import as is, is that the original import would throw error below:
`Error [ERR_REQUIRE_ESM]: require() of ES Module /home/runner/work/kiota/kiota/vscode/microsoft-kiota/node_modules/is-online/index.js from /home/runner/work/kiota/kiota/vscode/microsoft-kiota/out/kiotaInterop/kiotaInstall.js not supported.
Instead change the require of index.js in /home/runner/work/kiota/kiota/vscode/microsoft-kiota/out/kiotaInterop/kiotaInstall.js to a dynamic import() which is available in all CommonJS modules.`

Reply:
Not sure if it's a limitation of the test runtime but given the tsconfig.json specifies that "compilerOptions": {"module": "CommonJS"}, I interpret this to mean any of given runtimes would expect transpiled code to have CommonJS syntax? Maybe @thewahome , @koros and @rkodev can give more insight on this.

Reply:
thanks for providing more context here, and implementing the change.
I'd like to extend the discussion since you've mentioned the test infrastructure is based off commonjs. Have you tried switching it to ESM? The reason why I'm asking is because mixing module types is added complexity, reduced reliability, and we might get different results between the tests and the actual runtime.

Reply:
The project infrastructure as a whole is based off commonjs as the tests and the project files share the same tsconfig https://github.com/microsoft/kiota/blob/main/vscode/microsoft-kiota/tsconfig.json#L4. I haven't modified this at all. All other extensions I have encountered are based off commonjs.

Reply:
ok can you please log another issue to migrate vscode extension + test infrastructure to ESM? We can schedule the work at a later date.

Reply:
https://github.com/microsoft/kiota/issues/5752